src/XrdHttp/XrdHttpReq.cc

```diff
@@ -145,6 +145,10 @@ int XrdHttpReq::parseLine(char *line, int len) {
     // We memorize the headers also as a string                                                                                                                                              
     // because external plugins may need to process it differently                                                                                                                          
     std::string ss = val;
+    if(ss.length() >= 2 && ss.substr(ss.length() - 2, 2) != "\r\n") {
```

Comment:
So, I am confused. If the recommendation is to allow both \r\n and simple \n, why is the added code ignoring the recommendation in RFC-2616 19.3? Is it to be consistent with how weirdly the Connection header value is parse? Anyway, looking closely at the pre-existing code (i.e. not part of this pull request), I can see lots of issues that may bite us in the future.

Reply:
The reason for that is that Brian said that we can also ignore the RFC-2616 19.3 and send a bad request in case we see header lines not finishing with `\r\n`: https://github.com/xrootd/xrootd/issues/2273#issuecomment-2122679734

Reply:
I think my comment was a bit more subtle -- but I'm OK with this implementation.  My comment suggested is to  permit `\n` only if the client consistently uses it (that is, no mixing of `\n` and `\r\n` in the same request).

In fact, I looked up RFC 7230 which is a 2014 update that obsoletes RFC 2616.  In [section 3.5](https://datatracker.ietf.org/doc/html/rfc7230#section-3.5), they say that clients *MAY* permit use of `\n`-only.  This is quite different than the prior, circa 1999, advice of *SHOULD* accept such requests.

So yes: I'm OK with saying we've evolved beyond the advice from the late 1990's and can assume clients correctly frame requests by default.  Especially if assuming so allows us to better detect malformed requests or client bugs (such as this one!).